### ISSUE TITLE
💅 `useUniqueElementIds`: false positive on components that use `id` prop not for DOM id

### ISSUE THREAD

**Issue by sterliakov** [biomejs/biome#6722]
### Environment information

```bash
CLI:
  Version:                      2.0.6
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.19.3
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.12.3

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0
```

### Rule name

useUniqueElementIds

### Playground link

https://biomejs.dev/playground/?indentStyle=space&quoteStyle=single&indentWidth=4&lintRules=useUniqueElementIds&enabledAssist=false&code=ZgB1AG4AYwB0AGkAbwBuACAAQwB1AHMAdABvAG0AZQByAEkAZAAoAHsAIABpAGQAIAB9ACkAIAB7AAoAIAAgACAAIAByAGUAdAB1AHIAbgAgADwAPgBDAHUAcwB0AG8AbQBlAHIAIAAjAHsAaQBkAH0APAAvAD4AOwAKAH0ACgAKAGUAeABwAG8AcgB0ACAAZABlAGYAYQB1AGwAdAAgAGYAdQBuAGMAdABpAG8AbgAgAEQAZQBtAG8AKAApACAAewAKACAAIAAgACAAcgBlAHQAdQByAG4AIAA8AEMAdQBzAHQAbwBtAGUAcgBJAGQAIABpAGQAPQAiADEAIgAgAC8APgA7AAoAfQAKAA%3D%3D&language=jsx&ruleDomains.react=all

### Expected result

No error.

I understand it's borderline impossible to avoid false positives like this, and simply allowing static IDs on custom elements is a bad idea. I propose to add a whitelist option to this rule. We have a precedent of such whitelists: for example, `useExhaustiveDependencies` can be configured to assume that outputs of some hooks are static (behave like `useState` setter) and should not be included in dependency lists. Supporting similar `allowlist` option accepting component names would be an ideal solution for me.

For context, this is important for me because of [`FormattedMessage`](https://formatjs.github.io/docs/react-intl/components/#usage) in `react-intl` (aka FormatJS) that uses `id` as a message identifier. I have hundreds of occurrences of this in the codebase, so ignoring every one of them inline is not an option, and I had to disable the rule.

I am ready to contribute a fix if you agree that this is important enough. We can also add a similar option to `useValidAriaRole` (#6687) as @ematipico suggested.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
